File: CS/DXDiagram.CustomShapeProperties/ShapeClasses.cs
using DevExpress.Utils.Serializing;
using DevExpress.Xpf.Diagram;

namespace DXDiagram.CustomShapeProperties {
    public class DiagramShapeEx : DiagramShape {
        [XtraSerializableProperty]
        public string Description { get; set; }
        static DiagramShapeEx() {
            DiagramControl.ItemTypeRegistrator.Register(typeof(DiagramShapeEx));
        }
    }
}

File: VB/DXDiagram.CustomShapeProperties/ShapeClasses.vb
Imports DevExpress.Utils.Serializing
Imports DevExpress.Xpf.Diagram

Namespace DXDiagram.CustomShapeProperties

    Public Class DiagramShapeEx
        Inherits DiagramShape

        <XtraSerializableProperty>
        Public Property Description As String

        Shared Sub New()
            DiagramControl.ItemTypeRegistrator.Register(GetType(DiagramShapeEx))
        End Sub
    End Class
End Namespace
